yt_idefix/data_structures.py

```diff
@@ -576,9 +576,26 @@ def _parse_definitions_header(self) -> None:
             elif (unit_match := re.fullmatch(unit_regexp, line)) is not None:
                 unit = unit_match.group(1).lower() + "_unit"
                 expr = unit_match.group(2)
+                # Before evaluating the expression, replace the input parameters,
+                # pre-defined constants, code units and sqrt function
+                # that cannot be resolved. The order doesn't matter.
+                expr = re.sub(r"g_inputParam\[(\w+)\]", self._get_input_parameter, expr)
+                expr = re.sub(r"CONST_\w+", self._get_constants, expr)
+                expr = re.sub(r"UNIT_(\w+)", self._get_unit, expr)
```

Comment:
I *think* the group serves no purpose here, does it ?
```suggestion
                expr = re.sub(r"UNIT_\w+", self._get_unit, expr)
```

Reply:
I need the group in `self._get_constants()`, instead of the whole match, same as above.

Reply:
fair enough !